docs/testing.md

```diff
@@ -0,0 +1,179 @@
+- **Type checking** ensures that the construct you’re passing to a function matches what the function was designed to accept, preventing passing a string to a counting function that expects an integer, for instance.
+
+React Native comes with two such tools configured out of the box: [ESLint](https://eslint.org/) for linting and [Flow](https://flow.org/en/docs/) for type checking. You can also use [TypeScript](https://www.typescriptlang.org/), which is a typed language that compiles to plain JavaScript.
+
+## Writing Testable Code
+
+To start with tests, you first need to write code that is testable. Consider a car’s engine. For some reason it won’t start. In older cars, figuring out what is wrong with the engine requires the mechanic to take the engine apart, piece by piece, until they can figure out the issue. In modern cars, engines are outfitted with sensors that keep an eye on parts of the system, determining the exact location of part failures—sometimes before the engine even breaks down! Sensors changed the way engines were designed and manufactured.
+
+Similarly, with software, writing your code in many small functions that you can test (like adding sensors to the different components of an engine) is more effective than writing your entire program in one file with many lines of code (where you can only test if the engine starts or not). In this way, writing testable code is intertwined with writing clean, modular code. We encourage you to explore this topic in other learning resources.
+
+To help make your app more testable, it’s a good idea to separate the view part of your app, React components, from business logic and app state (regardless of whether you use Redux, MobX or other solutions). This way, you can separate tests of your business logic — which shouldn’t rely on your React components — from the components themselves, whose job is primarily rendering your app’s UI!
+
+Theoretically, you could go so far as to move all logic and data fetching out of your components. This way your components would be solely dedicated to rendering, and your state would be independent on your components - your app’s logic would work without any React components at all!
+
+After writing testable code, it’s time to write some actual tests! (Or if you do test-driven development, you actually write tests first!).
+
+React Native apps come with [Jest](https://jestjs.io) out of the box, ready to use. Jest is a popular JavaScript Testing Framework also started at Facebook. You can use it to write all types of tests featured in this guide. Let’s get started!
+
+### Structuring Tests
+
+Your tests should be short and ideally test only one thing. Let's start with an example unit test:
```

Comment:
Should this say something like "start with an example unit test written with Jest"?

Reply:
did that